diff --git a/node_modules/@expo/cli/build/src/start/doctor/ngrok/NgrokResolver.js b/node_modules/@expo/cli/build/src/start/doctor/ngrok/NgrokResolver.js
index 7a6faaf..3b68e0c 100644
--- a/node_modules/@expo/cli/build/src/start/doctor/ngrok/NgrokResolver.js
+++ b/node_modules/@expo/cli/build/src/start/doctor/ngrok/NgrokResolver.js
@@ -26,8 +26,12 @@ class NgrokResolver extends _ExternalModule.ExternalModule {
     }
 }
 function isNgrokClientError(error) {
-    var _error_body;
-    return error == null ? void 0 : (_error_body = error.body) == null ? void 0 : _error_body.msg;
+    try {
+        var _error_body;
+        return error == null ? void 0 : (_error_body = error.body) == null ? void 0 : _error_body.msg;
+    } catch {
+        return false;
+    }
 }
 
 //# sourceMappingURL=NgrokResolver.js.map
\ No newline at end of file
diff --git a/node_modules/@expo/cli/build/src/start/server/AsyncNgrok.js b/node_modules/@expo/cli/build/src/start/server/AsyncNgrok.js
index 4c42029..69f4fc9 100644
--- a/node_modules/@expo/cli/build/src/start/server/AsyncNgrok.js
+++ b/node_modules/@expo/cli/build/src/start/server/AsyncNgrok.js
@@ -204,7 +204,7 @@ class AsyncNgrok {
             const urlProps = await this._getConnectionPropsAsync();
             const url = await instance.connect({
                 ...urlProps,
-                authtoken: NGROK_CONFIG.authToken,
+                authtoken: process.env.NGROK_AUTHTOKEN || NGROK_CONFIG.authToken,
                 configPath,
                 onStatusChange (status) {
                     if (status === 'closed') {
@@ -221,8 +221,8 @@ class AsyncNgrok {
                 if ((0, _NgrokResolver.isNgrokClientError)(error)) {
                     var _error_body_details;
                     throw new _errors.CommandError('NGROK_CONNECT', [
-                        error.body.msg,
-                        (_error_body_details = error.body.details) == null ? void 0 : _error_body_details.err,
+                        error.body == null ? void 0 : error.body.msg,
+                        (_error_body_details = error.body == null ? void 0 : error.body.details) == null ? void 0 : _error_body_details.err,
                         _chalk().default.gray('Check the Ngrok status page for outages: https://status.ngrok.com/')
                     ].filter(Boolean).join('\n\n'));
                 }
@@ -233,7 +233,7 @@ class AsyncNgrok {
                 assertNgrok();
             }
             // Attempt to fix the issue
-            if ((0, _NgrokResolver.isNgrokClientError)(error) && error.body.error_code === 103) {
+            if ((0, _NgrokResolver.isNgrokClientError)(error) && (error.body == null ? void 0 : error.body.error_code) === 103) {
                 // Assert early if a custom subdomain is used since it cannot
                 // be changed and retried. If the tunnel subdomain is a boolean
                 // then we can reset the randomness and try again.
